src/exojax/spec/moldb.py

```diff
@@ -753,7 +753,13 @@ def Qr_HAPI(self, Tarr):
         allT = list(np.concatenate([[self.Tref], Tarr]))
         Qrx = []
         for iso in self.uniqiso:
-            Qrx.append(hapi.partitionSum(self.molecid, iso, allT))
+            Tmin, Tmax = hapi.get_TMIN_TMAX_FOR_BD_TIPS_2017_PYTHON(self.molecid, iso)
+            if(max(allT) > Tmax):
+                print('Warning: partition function ratio for isotope %d of molecule %d at temperatures > %.1fK is assumed to be the value at %.1fK since HAPI does not support those temperatures.' % (iso, self.molecid, Tmax, Tmax))
```

Comment:
sorry to insist but users could be making an extreme error by doing this. 

Partition functions cannot be clipped. As you can see running the code below, they are diverging fast with temperature :

```python

from radis.levels.partfunc import PartFuncHAPI
import numpy as np

T = np.linspace(300, 9000)

Q = PartFuncHAPI("CO", I=1)
Z = []
for Ti in T:
    Z.append(Q.at(T=Ti))
    
import matplotlib.pyplot as plt
plt.plot(T, Z)
plt.xlabel('Temperature')
plt.ylabel('Partition Function')
```
![image](https://user-images.githubusercontent.com/16088743/163378926-95865383-d0eb-4999-9b8c-5a8191cd5555.png)



 Having only a print() notice is dangerous --> no-one reads them in a batch computation.
I would really raise an error, and maybe, let the user consciously turn on a parameter (USE_OUT_OF_RANGE_PARTITION_FUNCTION)  to allow this behavior.

Then it's the users's problem and not the fault of the exojax developers ;)